<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
	<head>
		<meta name="generator" content="HTML Tidy, see www.w3.org">
		<meta http-equiv="content-type" content="text/html;charset=iso-8859-1">

		<title>Features of ermineJ</title>
		<link href="style.css" rel="stylesheet" type="text/css" media="all">
	</head>

	<body bgcolor="#ffffff">
		<h1>
			<img height="112" alt="ermineJ logo" src="images/logo1small.gif"
				width="266" border="0">
		</h1>

		<h1>
			Features of ermineJ
		</h1>

		<p>
			ErmineJ performs analyses of
			<b>gene sets</b> in
			<b>high-throughput genomics data</b> such as gene expression
			profiling studies. A typical goal is to determine whether particular
			biological pathways are "doing something interesting" in an
			experiment that generates long lists of candidates. The software is
			designed to be used by biologists with little or no informatics
			background. A command-line interface is available for users who wish
			to script the use of ermineJ.
		</p>

		<p>
			Major features include:
		</p>

		<ul>
			<li>
				Implementation of multiple methods for gene set analysis:

				<ul>
					<li>
						<a href="tutorial/ora.html">Over-representation analysis</a>
					</li>
					<li>
						<a href="tutorial/resampling.html">A resampling-based method
							that uses gene scores</a>
					</li>
					<li>
						<a href="tutorial/roc.html">A rank-based method that uses gene
							scores</a>
					</li>
					<li>
						A resampling-based method that uses
						<a href="tutorial/correlation.html">correlation between gene
							expression profiles</a> (a type of "cluster-enrichment" analysis).
					</li>
				</ul>
			</li>
			<li>
				Gene sets receive statistical scores (p-values), and
				<a href="tutorial/mtc.html">multiple test correction</a> is
				supported.
			</li>
			<li>
				Support of the
				<a href="http://www.geneontology.org">Gene Ontology</a> terminology;
				users can choose which aspects to analyze.
			</li>
			<li>
				Provides information on
				<a href="tutorial/multfiunc.html">gene multifunctionality</a>, which
				has and important impact on the interpretation of results.
			</li>
			<li>
				User files use
				<A href="formats/go.html">simple text formats</A>.
			</li>
			<li>
				Users can
				<A href="tutorial/classes.html">modify gene sets or create new
					ones</a>, allowing the use of arbitrary non-GO schemes such as
				<a href="http://www.genome.jp/kegg">KEGG</a> or
				<a href="http://www.reactome.org">Reactome</a>.
			</li>
			<li>
				The results can be
				<a href="tutorial/visualize.html">visualized within the software</a>.
			</li>
			<li>
				It is simple to compare multiple analyses of the same data set with
				different settings.
			</li>
			<li>
				<a href="tutorial/setgeneurl.html">User-definable hyperlinks</a> are
				provided to external sites to allow more efficient browsing of the
				results.
			</li>
			<li>
				For programmers, there is a
				<a href="tutorial/cli.html">command line interface</a> as well as a
				<a href="tutorial/Api.html">simple application programming
					interface</a> that can be used to plug ermineJ functionality into your
				own code
			</li>
		</ul>

		<p>
			<a href="license.html">License information</a>
		</p>



		<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl."
		: "http://www.");
document
		.write(unescape("%3Cscript src='"
				+ gaJsHost
				+ "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
		<script type="text/javascript">
var pageTracker = _gat._getTracker('UA-4826357-1');
pageTracker._initData();
pageTracker._trackPageview();
</script>
	</body>
</html>
