<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
	<head>
		<meta name="generator" content="HTML Tidy, see www.w3.org">

		<title>Frequently asked questions</title>
		<link rel="stylesheet" type="text/css" href="style.css">
	</head>

	<body>
		<IMG src="images/logo1small.gif" alt="">


		<h1>
			Frequently asked questions
		</h1>

		<p>
			<strong>Which gene scoring method should I use?</strong> If you are
			looking at differential expression (the usual case), we recommend
			using the
			<a href="tutorial/resampling.html">gene score resampling</a> or the
			<a href="tutorial/roc.html">ROC</a> methods.
			<a href="tutorial/ora.html">ORA</a> is most appropriate when you have
			a natural threshold for gene selection, such as "on chromosome 2".
			Otherwise, changing the threshold can alter the results, sometimes
			dramatically. If you are interested in clustering genes, the
			<a href="tutorial/correlation.html">correlation</a> method might be
			appropriate. There is more information on choosing methods on the
			help pages for each method.
		</p>

		<p>
			<strong>Should I run all methods and combine the results?</strong> We
			don't really recommend running all the methods and looking for
			overlaps, though running multiple methods and comparing the results
			might be useful. Familiarize yourself with the different methods to
			be aware of their properties. It is particularly important to
			recognize that
			<a href="tutorial/correlation.html">correlation analysis</a> works
			very differently from the other methods, and ORA analysis results can
			be sensitive to the threshold used.
		</p>

		<p>
			<strong>How should I deal with up-regulated vs.
				down-regulated genes? I want them analyzed separately.</strong> To make your
			input p-values "aware" of the direction of change, you typically need
			to use a
			<a href="http://en.wikipedia.org/wiki/Two-tailed_test">one-tailed
				test</a> to generate your p-values. ErmineJ doesn't handle that
			internally, you have to provide your own pvalues. So there would be
			one list of pvalues for up-regulation and one for down, using one
			tail and then the other of the distribution from which p-values are
			determined. Alternatively you can use the fold-changes. However, our
			recommendation is that you use p-values, especially if that is how
			you are ranking genes for other parts of your project. But some
			people use fold change. You have to decide which will best answer
			your question. If you think fold-change is a better representation of
			the ranking of genes, then use fold change.
		</p>


		<p>
			<STRONG>The messages at the bottom of the window go by too
				fast to read. </STRONG>You can read all messages (and other information) in
			the log file. You can view the log file from within ermineJ by
			selecting the "help" menu, choose "view log". The log file for the
			current or last run of ermineJ&nbsp;is stored on your hard drive in a
			file called ermineJ.log, it should be in your home directory.
		</p>
		<P>
			<STRONG>All my gene set p-values are coming out as zero.
				What is wrong?</STRONG> This can happen if you have not set the parameters
			for the analysis correctly. In particular, if your input file
			contains raw p-values, make sure the "larger scores are better" box
			is
			<strong>unchecked</strong> in the last analysis wizard step.
		</P>

		<p>
			<strong>What is the meaning of the "score" for each gene
				set?</strong> During analysis, each gene set is given a score which, along
			with the size of the gene set, is used to determine the statistical
			significance of the gene set. The meaning of this score depends on
			the type of analysis and is explained
			<a href="tutorial/scores.html">here</a>.
		</p>

		<p>
			<strong>How do I know if a gene set p value is
				'significant'?</strong> One has to consider the problem of multiple testing,
			but because the gene sets (Gene Onotology group) overlap, simple
			methods will be too conservative. For this reason we don't recommend
			using a Bonferroni correction, though if a Bonferonni correction
			yields 'significant' results, you can certainly consider them
			reasonably trustworthy.
		</p>

		<p>
			The software implements a false discovery rate (FDR) algorithm that
			leads to p values shown in a color at different levels of false
			discovery rate (see this
			<A href="tutorial/mtc.html">page</A>). However, this method still
			assumes the gene sets are independent and will lead to a convervative
			estimate of the FDR.
		</p>

		<p>
			If you want to use Bonferroni, you can either use the command-line
			interface, or compute them yourself.&nbsp;To compute the
			Bonferroni-corrected p values, multiply your gene set p values by the
			number of gene sets analyzed (maximum of 1). Then you can use a
			threshold of 0.05 (which would not otherwise be reasonable).
		</p>

		<p>
			<strong>What is multifunctionality?</strong> This refers to the fact
			that some genes have many more GO terms than others. This has a
			dramatic effect on the type of analysis ErmineJ does. This topic is
			covered in more detail in the
			<a href="tutorial/multifunc.html">manual</a>.
		</p>

		<p>
			<strong>My data set has no "significant" genes, but when I
				run ermineJ with resampling I get significant gene sets. How do I
				interpret these results?</strong> Carefully. ErmineJ's resampling methods are
			based on analyzing your gene expression data in isolation, without
			reference to any theoretical background model (other methods may be
			added in future versions of ermineJ that allow this). It is capable
			of identifying the most interesting gene sets in your data, but this
			does not necessarily mean those gene sets are interesting in general.
			We recommend using such findings as an exploratory method for helping
			identify potential genes of interest, but because no gene is
			statistically significant on its own, the results for any given gene
			would have to be dealt with on a case-by-case basis using
			considerations other that statistical significance.
		</p>

		<p>
			For a gene set to be considered statistically significant outside the
			context of your data set, one reasonable null hypothesis would yield
			a distribution of p-values in the gene set that is uniform (though
			this assumes the genes are independent). A uniform distribution of
			p-values is displayed in the
			<a href="tutorial/visualize.html">details view</a> of each class.
			When the blue line (your p-values) is far to the right of the
			expected distribution under the null (grey line), you can have higher
			confidence in the importance of that gene set. The reason we do not
			use this method exclusively is that in data sets where many genes
			have changed, every gene set will appear significant. Reasonable
			interpretation of the results requires consideration of both types of
			analyses.
		</p>

		<p>
			A related issue is that gene sets whose members have correlated
			transcription in all conditions (that is, not relevant to your study)
			can sometimes yield incorrect results. The remedy to this problem is
			to do resampling over the samples, but ermineJ currently does not
			support this ability. In the meantime, when you have weak or no clear
			differential expression effect on the genes in a gene set, and the
			genes are highly correlated, you should view the results carefully.
			Typically this affects gene sets like the ribosomal proteins,
			histones&nbsp;or proteasome, whose members are always highly
			correlated. You can test for gene sets like this using
			<a href="tutorial/correlation.html">correlation scoring</a>, though
			this seems likely to give helpful results only if the differential
			expression effects in your experiment are weak. Otherwise, gene sets
			which have strong "interesting" differential expression will also
			appear strongly correlated (as they should be).
		</p>

		<p>
			Data sets where there are weak signals can also be heavily influenced
			by
			<a href="tutorial/multifunc.html">multifunctionality</a>.
		</p>

		<p>
			<strong>Do I input the list of gene scores for the
				significantly changed genes, or all the genes?</strong> All of them. The
			resampling method uses the gene scores for all the genes. The
			built-in overrepresentation analysis will use the threshold you set
			in the gene score threshold parameter to select genes. Note that we
			find that removing unexpressed genes can help the analysis, but it
			isn't required. Note that if your data have been filtered, the output
			panel does not list the actual number of probes or genes in your gene
			sets. See
			<A href="tutorial/numitems.html">this page</A> &nbsp;for a
			clarification.
		</p>

		<p>
			<strong>How do I find out which genes are in a gene set
				without doing an analysis?</strong> You can use the "View/Modify Gene set"
			tools, available from the main 'analysis' menu, or the
			context-specific pop-up menu on the gene set list. See
			<A href="tutorial/classes.html#viewgeneset">this page</A>
		</p>

		<p>
			<strong>When I open the output file in Notepad it looks like
				a mess of letters and numbers. How do I view this file?</strong> The output
			will not look right when viewed in a text editor or a web browser. It
			will look correct in Excel if you open it as a tab delimited file.
		</p>

		<p>
			<strong>My excel spreadsheet with my gene scores isn't being
				accepted as input.</strong> You cannot input Excel spreadsheets; you have to
			save them as text first. See
			<A href="formats/excel-astext.html">this page</A> for details.
		</p>

		<p>
			<strong>I don't see an annotation file for my microarray.
				What should I do?</strong> Unfortunately we cannot generally create
			annotations for microarrays that are not commonly used. If we
			identify a commonly used platform, we will add it. This means that
			'home made' spotted microarrays are not supported in most cases. Feel
			free to let us know if you have a microarray design that needs
			annotations for class scoring. In the meantime, you could try to use
			one of the "Generic" annotation files we provide, possibly with the
			help of your local friendly bioinformatician.
		</p>

		<p>
			<strong>When doing an analysis, why did I see a warning
				"Attempt to take the log of a non-positive value"?</strong> If your gene
			score file contains negative or zero values, and you check the
			"negative log-transform gene scores" box, you will see this error. If
			this concerns you, you should clean up your gene score file. If not,
			ermineJ sets these values to a small number (10
			<sup>
				-15
			</sup>
			).
		</p>

		<p>
			<strong>When doing an analysis, why did I see a warning "<font
				color="#2a00ff" size="1"><font color="#000000" size="4">"Some
						probes in your gene score file don't match the ones in the
						annotation file"</font> </font> "?</strong> This happens if your gene annotation file
			isn't the right one for the microarray you used (in which case your
			analysis will not work well), but can also happen if your file format
			isn't quite right. The analysis still proceeds when this happens.
		</p>

		<p>
			<strong>When doing an analysis, why did I see a warning"<font
				color="#2a00ff" size="1"><font color="#000000" size="4">"Non-numeric
						gene scores(s) ( '#NUM!' ) found for input file. These are set to
						an initial value of zero."</font> </font> ", or something similar?</strong> This warning
			appears if the column you selected for your gene score file contains
			non-numeric data. This can happen if you have missing or invalid
			values (sometimes appearing as '#NUM!' in Microsoft Excel), but can
			also happen if you have chosen the wrong column in your data. Because
			such values are interpeted as zeros, you should be careful that your
			file contains the data you want. If you want to avoid problems you
			should remove those rows from your file entirely.
		</p>

		<p>
			<strong>The resampling is taking too long, any suggestions?</strong>
			This problem primarily affects the correlation score analysis in
			particular, which is very computationally intensive.
		</p>

		<p>
			There are few ways to speed things up. One is to
			<strong>uncheck</strong>
			<font color="#000000">the "</font><font color="#2a00ff" size="1"><font
				color="#000000" size="4"><font size="3"><em>Always
							use full resampling</em> </font>" <font size="3">checkbox in the
						analysis setup wizard. This enables approximations which, while
						potentially yielding less accurate p-values, greatly increase
						processing speed. The results you get with this box unchecked will
						be reasonably similar to what you would get with full resampling.</font>
			</font> </font>
		</p>

		<p>
			Another tip is to not look at very large values for the "maximum
			class size" value. We usually use a value on the order of 200.
		</p>

		<p>
			Finally, you can reduce the number of iterations. You might try
			temporarily setting this to 10000 while you experiment with the other
			settings, and then, once you find a setting you like, you can crank
			up the iterations for the "final" run (for example to 200000, which
			still should not take very long for the gene-score based resampling).
		</p>

		<p>
			<strong>Why do my resampling results vary slightly from run
				to run?</strong> This happens either when too few iterations are run, or
			sometimes when using the approximation methods. To ensure the highest
			accuracy, set the number of iterations to a larger value (say,
			200,000 for gene score resampling) and
			<strong>check</strong> the
			<em>Always use full resampling</em><font size="4">"</font>
			<font size="3">checkbox in the analysis setup wizard.</font>
		</p>

		<p>
			<strong>What happens to probes that don't have any GO
				annotations?</strong> They are ignored.
		</p>

		<p>
			<strong>For resampling analysis of gene scores, how is the
				gene score threshold determined?</strong> This is a trick question. There is
			no threshold used. Instead, all genes in a set contribute to the
			score for the gene set. For more details on how this works, see
			<A href="tutorial/resampling.html">this page</A> .
		</p>

		<p>
			<strong>What happens if there are two probes for&nbsp;one
				gene?</strong> See
			<A href="tutorial/genereps.html">this page</A> .
		</p>

		<p>
			<strong>Why are the number of gene and probes shown in the
				output panel incorrect when my data set has been filtered?</strong> See
			<a href="tutorial/numitems.html">this page</a>.
		</p>

		<p>
			<strong>What do the different icons mean in the tree view?</strong>
			The icons are used to indicate where to look to find "significant"
			gene sets. The specific icons are explained
			<A href="tutorial/analysis.html#treepanel">here</A>
		</p>

		<p>
			<strong>What do the different colors mean in the table and
				tree views?</strong> The colors indicate different levels of statistical
			significance, as explained
			<A href="tutorial/mtc.html">here</A>
		</p>
		<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl."
		: "http://www.");
document
		.write(unescape("%3Cscript src='"
				+ gaJsHost
				+ "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
		<script type="text/javascript">
var pageTracker = _gat._getTracker('UA-4826357-1');
pageTracker._initData();
pageTracker._trackPageview();
</script>
	</body>
</html>
