<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
<head>
<meta name="generator" content="HTML Tidy, see www.w3.org">

<title>Frequently asked questions</title>

<link rel="stylesheet" type="text/css" href="style.css">
</head>

<body>
<img src="images/logo1small.gif" />
<h1>Frequently asked questions</h1>

<p><strong>What is the meaning of the "score" for each gene set?</strong>
During analysis, each gene set is given a score which, along with the
size of the gene set, is used to determine the statistical significance
of the gene set. The meaning of this score depends on the type of
analysis and is explained <a href="scores.html">here</a>.</p>

<p><strong>How do I know if a gene set p value is 'significant'?</strong>
One has to consider the problem of multiple testing, but because the
gene sets (Gene Onotology group) overlap, simple methods will be too
conservative. For this reason we don't recommend using a Bonferroni
correction, though if a Bonferonni correction yields 'significant'
results, you can certainly consider them reasonably trustworthy.</p>

<p>The software implements a false discovery rate (FDR) algorithm that
leads to p values shown in a color at different levels of false
discovery rate (see this <A href="tutorial/mtc.html">page</A>). However,
this method still assumes the gene sets are independent and will lead to
a convervative estimate of the FDR.</p>

<p>If you want to use Bonferroni, you can either use the command-line
interface, or compute them yourself.&nbsp;To compute the
Bonferroni-corrected p values, multiply your gene set p values by the
number of gene sets analyzed (maximum of 1). Then you can use a
threshold of 0.05 (which would not otherwise be reasonable).</p>

<p><STRONG>My data set has no "significant" genes, but when I run
ermineJ with resampling I get significant gene sets. How do I interpret
these results?</STRONG> Carefully. ErmineJ's resampling methods are
based on analyzing your gene expression data in isolation, without
reference to any theoretical background model (other methods may be
added in future versions of ermineJ that allow this). It is capable of
identifying the most interesting gene sets in your data, but this does
not necessarily mean those gene sets are interesting in general. We
recommend using such findings as an exploratory method for helping
identify potential genes of interest, but because no gene is
statistically significant on its own, the results for any given gene
would have to be dealt with on a case-by-case basis using considerations
other that statistical significance.</p>

<p>For a gene set to be considered statistically significant outside the
context of your data set, a reasonable null hypothesis would yield a
distribution of p-values in the gene set that is uniform. A uniform
distribution of p-values is displayed in the<A
	href="tutorial/visualize.html"> details view</A> of each class. When
the blue line (your p-values) is far to the right of the expected
distribution under the null (grey line), you can have higher confidence
in the importance of that gene set. The reason we do not use this method
exclusively is that in data sets where many genes have changed, every
gene set will appear significant. Reasonable interpretation of the
results requires consideration of both types of analyses.</p>

<p><strong>Do I input the list of gene scores for the significantly
changed genes, or all the genes?</strong> All of them. The resampling
method uses the gene scores for all the genes. The built-in
overrepresentation analysis will use the threshold you set in the gene
score threshold parameter to select genes. Note that we find that
removing unexpressed genes can help the analysis, but it isn't required.
Note that if your data have been filtered, the output panel does not
list the actual number of probes or genes in your gene sets. See <A
	href="tutorial/numitems.html">this page</A> &nbsp;for a clarification.</p>

<p><strong>How do I find out which genes are in a gene set without doing
an analysis?</strong> You can use the "View/Modify Gene set" tools,
available from the main 'analysis' menu, or the context-specific pop-up
menu on the gene set list. See <a
	href="tutorial/classes.html#viewgeneset">this page</a></p>

<p><strong>When I open the output file in Notepad it looks like a mess
of letters and numbers. How do I view this file?</strong> The output
will not look right when viewed in a text editor or a web browser. It
will look correct in Excel if you open it as a tab delimited file.</p>

<p><strong>My excel spreadsheet with my gene scores isn't being accepted
as input.</strong> You cannot input Excel spreadsheets; you have to save
them as text first. See <A href="excel-astext.html">this page</A> for
details.</p>

<p><strong>I don't see an annotation file for my microarray. What should
I do?</strong> Unfortunately we cannot generally create annotations for
microarrays that are not commonly used. If we identify a commonly used
platform, we will add it. This means that 'home made' spotted
microarrays are not supported in most cases. Feel free to let us know if
you have a microarray design that needs annotations for class scoring.
In the meantime, you could try to use one of the "Generic" annotation
files we provide, possibly with the help of your local friendly
bioinformatician.</p>

<p><strong>When doing an analysis, why did I see a warning "Attempt to
take the log of a non-positive value"?</strong> If your gene score file
contains negative or zero values, and you check the "negative
log-transform gene scores" box, you will see this error. If this
concerns you, you should clean up your gene score file. If not, ermineJ
sets these values to a small number (10<sup>-15</sup> ).</p>

<p><strong>When doing an analysis, why did I see a warning "<font
	color="#2a00ff" size="1"><font color="#000000" size="4">"Some probes in
your gene score file don't match the ones in the annotation file"</font></font>
"?</strong> This happens if your gene annotation file isn't the right
one for the microarray you used (in which case your analysis will not
work well), but can also happen if your file format isn't quite right.
The analysis still proceeds when this happens.</p>

<p><strong>When doing an analysis, why did I see a warning"<font
	color="#2a00ff" size="1"><font color="#000000" size="4">"Non-numeric
gene scores(s) ( '#NUM!' ) found for input file. These are set to an
initial value of zero."</font></font> ", or something similar?</strong>
This warning appears if the column you selected for your gene score file
contains non-numeric data. This can happen if you have missing or
invalid values (sometimes appearing as '#NUM!' in Microsoft Excel), but
can also happen if you have chosen the wrong column in your data.
Because such values are interpeted as zeros, you should be careful that
your file contains the data you want. If you want to avoid problems you
should remove those rows from your file entirely.</p>

<p><strong>The resampling is taking too long, any suggestions?</strong>
This problem primarily affects the correlation score analysis in
particular, which is very computationally intensive.</p>

<p>There are few ways to speed things up. One is to <strong>uncheck</strong>
<font color="#000000">the "</font><font color="#2a00ff" size="1"><font
	color="#000000" size="4"><font size="3"><em>Always use full resampling</em></font>"
<font size="3">checkbox in the analysis setup wizard. This enables
approximations which, while potentially yielding less accurate p-values,
greatly increase processing speed. The results you get with this box
unchecked will be reasonably similar to what you would get with full
resampling.</font></font></font></p>

<p>Another tip is to not look at very large values for the "maximum
class size" value. We usually use a value on the order of 200.</p>

<p>Finally, you can reduce the number of iterations. You might try
temporarily setting this to 10000 while you experiment with the other
settings, and then, once you find a setting you like, you can crank up
the iterations for the "final" run (for example to 200000, which still
should not take very long for the gene-score based resampling).</p>


<P><STRONG>Why do my resampling results vary slightly from run to run?</STRONG>
This happens either when too few iterations are run, or sometimes when
using the approximation methods. To ensure the highest accuracy, set the
number of iterations to a larger value (say, 200,000 for gene score
resampling) and <STRONG>check</STRONG> the <EM>Always use full
resampling</EM><FONT size="4">" </FONT><FONT size="3">checkbox in the
analysis setup wizard.</FONT></P>

<p><strong>What happens to probes that don't have any GO annotations?</strong>
They are ignored.</p>

<p><strong>For resampling analysis of gene scores, how is the gene score
threshold determined?</strong> This is a trick question. There is no
threshold used. Instead, all genes in a set contribute to the score for
the gene set. For more details on how this works, see <A
	href="tutorial/resampling.html">this page</A> .</p>

<p><strong>What happens if there are two probes for&nbsp;one gene?</strong>
See <A href="tutorial/genereps.html">this page</A> .</p>

<p><strong>Why are the number of gene and probes shown in the output
panel incorrect when my data set has been filtered?</strong> See <A
	href="tutorial/numitems.html">this page</A>.</p>
</body>
</html>
