<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>
<head>
<meta name="generator" content="HTML Tidy, see www.w3.org">

<title>Class Scoring</title>
<link type="text/css" rel="stylesheet" href="css/default.css">
</head>

<body bgcolor="#ffffff" link="#800000" text="#000000">
<table border="0" cellpadding="2" cellspacing="3" width="776">
	<tr>
		<td width="72"><!-- Logo space --> <A href="/pavlidis">Home</A></td>

		<td width="687"><!-- Header space -->

		<h1><IMG height="112" alt="ermineJ logo" src="images/logo1small.gif"
			width="266" border="0"></h1>

		<h1>Gene Ontology analysis for&nbsp;microarray data</h1>
		</td>
	</tr>

	<tr>
		<td width="72"><!-- Left gutter space --> <IMG height="1" alt=""
			src="/images/shim.gif" width="70" border="0"></td>

		<td valign="top" width="687">
		<h2>Overview</h2>

		<p>ErmineJ performs analyses of <b>gene sets</b> in <b>expression
		microarray data</b>. A typical goal is to determine whether particular
		biological pathways are "doing something interesting" in the data. The
		software is designed to be used by biologists with little or no
		informatics background. A command-line interface is available for
		users who wish to script the use of ermineJ.</p>

		<p>Key features include:</p>

		<ul>
			<li>Implementation of several methods for gene set analysis:

			<ul>
				<li><A href="ermineJ-help/html/tutorial/ora.html">Over-representation
				analysis</A>
				<li><A href="ermineJ-help/html/tutorial/resampling.html">A
				resampling-based method that uses gene scores</A>
				<li>A resampling-based method that uses <a
					href="ermineJ-help/html/tutorial/correlation.html">correlation
				between gene expression profiles</a> .</li>
			</ul>
			<li>Gene sets receive statistical scores (p-values), and multiple
			test correction is supported.
			<li>Support of the Gene Ontology terminology
			<li>User files use <A href="ermineJ-help/html/formats/go.html">simple
			text formats</A>.
			<li>Users can <A href="ermineJ-help/html/tutorial/classes.html">modify
			gene sets or create new ones</A>.
			<li>The results can be <a
				href="ermineJ-help/html/tutorial/visualize.html">visualized within
			the software</a>.
			<li>It is simple to compare multiple analyses of the same data set
			with different settings.
			<li>Hyperlinks are provided to external sites to allow more efficient
			browsing of the results.</li>
		</ul>

		<p><A href="ermineJ-help/html/license.html">License information</A> -
		ermineJ is free software.</p>

		<p>There are some <a href="screenshots.html">screen shots</a> available,
		and more information is available in the <a href="ermineJ-help/html">manual</a>.
		
		<h2>News</h2>

		<p>Version 2.1 has been released. Many new improvements and changes.
		For details see the <a href="versions.html">version history</a></p>

		<p><A href="versions.html">Full version history</A></p>

		<h2>Run it now</h2>

		<p>You can <A href="webstart">run ermineJ</A> from any operating
		system that supports Java (in theory) without running the installer by
		using java webstart. If you have Webstart installed, you <a
			href="webstart/ermineJ.jnlp">run ermineJ right now</a>.</p>

		<p>This a good way to get the latest version, as we will keep the link
		updated with the most recent development build. On the other hand,
		this version should be considered less stable than the other releases.</p>

		<h2>Documentation</h2>

		<p>The software comes with on-line help. The manual is also <a
			href="ermineJ-help/html">available online</a> (webstart users will
		have to use this option for now). If you are having problems with the
		software please read the <a href="ermineJ-help/html/faq.html">FAQ</a>.
		If your question isn't answered in the FAQ or in the manual, feel free
		to <a href="mailto:pavlidis@dbmi.columbia.edu">contact us</a></p>
		<hr>

		<h2>Download and installation</h2>

		<h4>Requirements</h4>

		<p>ermineJ requires a Java 2 Runtime Environment (JRE) 1.4 or newer to
		run. You can download the platform-specific packages from SUN's
		website (<a href="http://java.sun.com/j2se/">http://java.sun.com/j2se/</a>
		). Users of MS Windows NT/2000/XP can alternatively download the
		Windows installer with bundled JRE package below. On machines with
		less than 256 Mb of RAM ermineJ may suffer from poor performance.</p>

		<h4>Software Files</h4>

		<p>Download the appropriate file for your platform:</p>

		<table width="100%" border="0" cellspacing="1" cellpadding="3"
			bgcolor="#aaaaaa">
			<tr bgcolor="#eeeeee">
				<td width="37%"><strong>Description</strong></td>

				<td width="30%"><strong>File</strong></td>

				<td width="33%"><strong>Size</strong></td>
			</tr>

			<tr>
				<td bgcolor="#ffffff">Windows installer with bundled JRE</td>

				<td bgcolor="#ffffff"><a href="distributions/ermineJ-setup-jre.exe">ermineJ-setup-jre.exe</a></td>

				<td bgcolor="#ffffff">~30 MB</td>
			</tr>

			<tr>
				<td bgcolor="#ffffff">Windows installer without JRE</td>

				<td bgcolor="#ffffff"><a href="distributions/ermineJ-setup.exe">ermineJ-setup.exe</a></td>

				<td bgcolor="#ffffff">~7 MB</td>
			</tr>

			<tr>
				<td bgcolor="#ffffff">Generic bundle</td>

				<td bgcolor="#ffffff"><a
					href="distributions/ermineJ-2.0.5-bindist.tar.gz">ermineJ-2.1-bindist.tar.gz</a></td>

				<td bgcolor="#ffffff">~7 MB</td>
			</tr>

			<tr>
				<td bgcolor="#ffffff">Source code</td>

				<td bgcolor="#ffffff"><u><a
					href="distributions/ermineJ-2.0.5-src.tar.gz"><font color="#800000">ermineJ-2.1-src.tar.gz</font></a></u></td>

				<td bgcolor="#ffffff">&lt; 1 MB</td>
			</tr>
		</table>

		<h3>Annotation files</h3>

		<p>To use the software, you have to have microarray annotation files
		and a recent Gene Ontology XML file. ErmineJ is distributed with
		sample files so most users can get started faster, but you will either
		need or want to get different files.</p>

		<p>Go <A href="/annots/">here</A> to get annotation files.&nbsp; Copy
		them somewhere you can find them. It is not necessary to unzip them.</p>

		<p>Go <a href="http://www.godatabase.org/dev/database/archive/latest/">here</a>
		to get the XML file (something like go_200503-termdb.rdf-xml.gz, found
		here: <a href="http://www.godatabase.org/dev/database/archive/latest/">http://www.godatabase.org/dev/database/archive/latest/</a>;
		the date portion of the name may have changed). It is not necessary to
		unzip the file.</p>

		<p>Note that you can define your own gene annotation files or use some
		<a href="ermineJ-help/html/formats/annotations.html">third-party file
		formats</a>. You can also <a
			href="ermineJ-help/html/tutorial/classes.html">define your own gene
		sets</a> using ermineJ or <a
			href="ermineJ-help/html/formats/geneset.html">import them into
		ermineJ</a>.</p>

		<h3>Installation</h3>

		<p>On MS Windows NT/2000/XP, simply run the installer.</p>

		<p>For other platforms, you should be able to use the "Generic
		bundle". There is a README file included that gives some instructions.</p>



		<h3>Contact</h3>

		<p>Contact <A href="mailto:pavlidis@dbmi.columbia.edu">Paul Pavlidis</A>
		or <A href="mailto:keshav@cu-genome.org">Kiran Keshav</A> with bug
		reports, feature requests, and questions.</p>

		<h3>Publications using the software or methods</h3>

		<ul>
			<li><!--StartFragment -->Sibille, E., Arango, V., Galfalvy, H.C.,
			Pavlidis, P., Ellis, S.P. and Mann, J.J. (2004) Genomic analysis of
			depression and suicide in human prefrontal cortex.
			Neuropsychopharmacology 29:351-361 .
			<li><!--StartFragment -->&nbsp;Pavlidis P., Qin J, Arango V, Mann J,
			Sibille E. (2004) Using the Gene Ontology for microarray data mining:
			A comparison of methods and application to age effects in human
			prefrontal cortex. Neurochemical Research, 29(6) 1213-1222&nbsp;.
			<li>Pavlidis, P., Qin, J and Sibille E (2003) Statistical analysis of
			Gene Ontology classes as tools for understanding gene expression
			changes in the brain. Society for neuroscience Annual Meeting
			Abstract 758.12. (<a href="neurosci2003.ppt"><font color="#0000ff">download
			the powerpoint version of the poster</font></a> )
			<li>Pavlidis, Lewis and Noble, Proceedings of the Pacific Symposium
			on Biocomputing 7. pp 474-485, 2002. (<a
				href="/pavlidis/doc/reprints/pavlidis-psb2002.pdf"><font
				color="#0000ff">Download a reprint</font></a> .</li>
		</ul>

		<h3>Credits</h3>

		<p>The software was written by Paul Pavlidis, Homin Lee, William
		Braynen, and Kiran Keshav. Some code is based on an earlier version of
		the software written by Shahmil Merchant and Edward Chen, based in
		turn on a Perl implementation by Paul Pavlidis. Our Gene Ontology
		annotations were obtained through the <a
			href="http://www.geneontology.org/">Gene Ontology Consortium web site</a>.</p>
		</td>
	</tr>
</table>
</body>
</html>
