<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>
	<head>
		<meta name="generator" content="HTML Tidy, see www.w3.org">

		<title>ErmineJ</title>
		<link type="text/css" rel="stylesheet" href="css/default.css">
	</head>

	<body bgcolor="#ffffff" link="#800000" text="#000000">
		<table border="0" cellpadding="2" cellspacing="3" width="776">
			<tr>
				<td width="2">
					<!-- Logo space -->
				</td>

				<td width="687">
					<!-- Header space -->
					<a href="/ermineJ"><IMG src="images/banner.jpg" border="0">
					</a>

					<h1>
						Gene set analysis for&nbsp;high-throughput data
					</h1>
					<P>
						<STRONG>Version 3.0</STRONG>
					</P>
				</td>
			</tr>
			<tr>
				<td></td>
				<td>
					<!--  navigation -->
					<table>
						<tr>
							<td>
								<a href="index.html">Home</a>
							</td>
							<td>
								&nbsp;&nbsp;
							</td>
							<td>
								<a href="downloadInstall.html">Download</a>
							</td>
							<td>
								&nbsp;&nbsp;
							</td>
							<td>
								<a href="ermineJ-help/classScore/html/">Help</a>
							</td>
							<td>
								&nbsp;&nbsp;
							</td>
							<td>
								<a href="publications.html">Pubs</a>
							</td>
							<td>
								&nbsp;&nbsp;
							</td>
							<td>
								<a href="screenshots.html">Screens</a>
							</td>
							<td>
								&nbsp;&nbsp;
							</td>
							<td>
								<a href="versions.html">Versions</a>
							</td>
							<td>
								&nbsp;&nbsp;
							</td>
							<td>
								<a href="contact.html">Contact</a>
							</td>
						</tr>
					</table>

				</td>
			</tr>
			<tr>
				<td width="72">
					<!-- Left gutter space -->
					<img height="1" alt="" src="/images/shim.gif" width="30" border="0">
				</td>

				<td valign="top" width="687">

					<h2>
						Overview
					</h2>
					
					<p><strong>Powerful</strong>, 
					<strong>flexible</strong> and <strong>simple</strong> gene set analysis.

					<p>
						ErmineJ performs analyses of
						<b>gene sets</b> in
						<b>high-throughput genomics data</b> such as gene expression
						profiling studies. A typical goal is to determine whether
						particular biological pathways are "doing something interesting"
						in an experiment that generates long lists of candidates. The
						software is designed to be used by biologists with little or no
						informatics background. A command-line interface is available for
						users who wish to script the use of ermineJ.
					</p>
					<p>
						A description of ermineJ was published as: Lee H.K., Braynen W.,
						Keshav K. and Pavlidis P. (2005) ErmineJ: Tool for functional
						analysis of gene expression data sets. BMC Bioinformatics 6:269.
						(Download a
						<a target="_blank" href="/pavlidis/reprints/ermineJ.pdf">PDF</a>
						or view the
						<a target="_blank"
							href="http://www.biomedcentral.com/1471-2105/6/269/abstract">abstract</a>).
					</p>
					<p>
						While ErmineJ has a full
						<a href="ermineJ-help/classScore/html/">user's manual</a>, there
						is a
						<a target="_blank"
							href="http://www.nature.com/nprot/journal/v5/n6/full/nprot.2010.78.html">Nature
							Protocols</a> paper that contains many additional tips and a detailed
						walk-through.
					</p>

					<p>
						ErmineJ can be
						<strong><a href="downloadInstall.html">downloaded
								here</a> </strong>, or
						<strong><a href="webstart/ermineJ.jnlp">run ErmineJ
								right now</a> </strong> using WebStart if you already have Java installed.
					</p>

					<p>
						Major features include:
					</p>

					<ul>
					
						<li>Has a simple-to-use graphical user interface; runs on your desktop (not browser-based).</li>

						<li>
							Implements multiple methods for gene set analysis:

							<ul>
								<li>
									Standard <a href="ermineJ-help/classScore/html/tutorial/ora.html">over-representation
										analysis</a>, requiring a threshold to select a "hit list" of genes.
								</li>
								<li>
									<a href="ermineJ-help/classScore/html/tutorial/resampling.html">A
										resampling-based method that uses gene scores</a>
								</li>
								<li>
									<a href="ermineJ-help/classScore/html/tutorial/roc.html">A
										rank-based method that uses gene scores</a>
								</li>
								<li>
									A resampling-based method that uses
									<a href="tutorial/correlation.html">correlation between
										gene expression profiles</a> (a type of "cluster-enrichment"
									analysis).
								</li>
							</ul>
						</li>
						<li> Not specific to any taxon, measurement type, or platform. While it was
							 originally designed with expression profiling in mind, it can be use for any genome-wide
							  analysis that yields rankings of genes, provided the genes have some gene sets assigned (such as GO annotations).
						</li>
						<li>
							Gene sets receive statistical scores (p-values), and
							<a href="ermineJ-help/classScore/html/tutorial/mtc.html">multiple
								test correction</a> is provided.
						</li>
						<li>
							Supports the
							<a href="http://www.geneontology.org">Gene Ontology</a>
							terminology; users can choose which aspects to analyze.
						</li>
						<li>
							Users can
							<A href="ermineJ-help/classScore/html/tutorial/classes.html">modify
								gene sets or create new ones</a>, allowing customization and the use of arbitrary
							non-GO gene groups including <a href="http://www.broadinstitute.org/gsea/msigdb">MolSigDB</a>,
							<a href="http://www.genome.jp/kegg">KEGG</a> or
							<a href="http://www.reactome.org">Reactome</a>.
						</li>
						<li>
							Provides information on
							<a href="ermineJ-help/classScore/html/tutorial/multifunc.html">gene
								multifunctionality</a>, important for the interpretation of results.
						</li>
						<li>
							The results can be
							<a href="ermineJ-help/classScore/html/tutorial/visualize.html">visualized
								within the software</a>.
						</li>
						<li>
							User files use
							<A href="ermineJ-help/classScore/html/formats/go.html">simple
								text formats</A>.
						</li>
						<li>
							It is simple to compare multiple analyses of the same data set
							with different settings.
						</li>
						<li>
							There is a (fairly) complete use manual.
						</li>
						<li>
							For programmers, there is a
							<a href="ermineJ-help/classScore/html/tutorial/cli.html">command
								line interface</a> as well as a
							<a href="ermineJ-help/classScore/html/tutorial/Api.html">simple
								application programming interface</a> that can be used to plug
							ermineJ functionality into your own code
						</li>
					</ul>

					<p>
						<A href="ermineJ-help/classScore/html/license.html">License
							information</A> - ermineJ is free software.
					</p>

					<p>
						There are some
						<A href="screenshots.html">screen shots</A> available, and more
						information is available in the
						<a href="ermineJ-help/classScore/html">manual</a>.
					</p>

					<h2>
						News
					</h2>

					<p>
						Version 3.0 - (August 2011)
					</p>
					<p>We are excited to announce a major new release of ErmineJ. There are a number of new features,
					and overall the software is improved and easier to use. Some highlights:</p>
					
					<ul>
						<li>
						Introduction of diagnostics for "gene multifunctionality". See
						<a href="ermineJ-help/classScore/html/tutorial/multifunc.html">the
							documentation</a> for details.
						</li>
						<li>Numerous user interface improvements have been made including filtering of displays and better error handling.</li>
						<li>Support is much improved for non-GO gene annotation schemes such as MolSigDB or KEGG</li>
						<li>Much of the code has been rewrittena and many minor and major bugs have been fixed. Note that while the
						core algorithms of ErmineJ have not been changed, changes to the way gene annotations are handled
						may cause differences in results.</li>
					</ul>
					
					<p>We believe this upgrade will increase the future maintainability of the software and sets the stage for the
					introduction of new features. We look forward to your feedback.</p>
					
					<p>
						For a more complete change list see the <a href="versions.html">version history</A>
					</p>

					<p>
						<a href="oldnews.html">Old news</a>
					</p>

					<h2>
						Documentation
					</h2>

					<p>
						The software comes with on-line help. The manual is also
						<a href="ermineJ-help/classScore/html">available online</a>
						(webstart users will have to use this option for now) or as an out-dated
						<a href="ermineJ-help/classScore/ermineJ-2.1.1-manual.pdf">PDF</a>.
						If you are having problems with the software please read the
						<a href="ermineJ-help/classScore/html/faq.html">FAQ</a>. If your
						question isn't answered in the FAQ or in the manual, feel free to
						<a href="contact.html">contact us</a>
					</p>

					<p>
						There is also a
						<a target="_blank"
							href="http://www.nature.com/nprot/journal/v5/n6/full/nprot.2010.78.html">Nature
							Protocols</a> paper describing how to use ErmineJ. It is meant to
						complement the
						<a href="ermineJ-help/classScore/html/">users manual</a>.
					</p>

					<p>
						For more details about ermineJ and how it is used, see the
						<a href="publications.html">publications list</a>.
					</p>

					<h2>
						Mailing list
					</h2>
					<p>
						If you use ermineJ, we urge you to subscribe to the
						<a
							href="http://lists.chibi.ubc.ca/mailman/listinfo/erminej-announce">ermineJ
							users mailing list</a>. This serves two purposes. First, it will
						allow you to automatically receive information on updates to
						ermineJ. Second, it helps us track how many people use ermineJ.
						This is especially important when we try to obtain support for
						ermineJ development.
					</p>
					<hr>
				</td>
			</tr>
		</table>

		<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl."
		: "http://www.");
document
		.write(unescape("%3Cscript src='"
				+ gaJsHost
				+ "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
		<script type="text/javascript">
var pageTracker = _gat._getTracker('UA-4826357-1');
pageTracker._initData();
pageTracker._trackPageview();
</script>

	</body>
</html>

