<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>
  <head>
    <meta name="generator" content="HTML Tidy, see www.w3.org">

    <title>Class Scoring</title>
    <link type="text/css" rel="stylesheet" href="css/default.css">
  </head>

  <body bgcolor="#ffffff" link="#800000" text="#000000">
    <table border="0" cellpadding="2" cellspacing="3" width="776">
      <tr>
        <td width="72"><!-- Logo space -->
        <a href="/pavlidis">Home</a></td>

        <td width="687">
          <!-- Header space -->

          <h1><img height="112" alt="ermineJ logo" src=
          "images/logo1small.gif" width="266" border="0"></h1>

          <h1>Gene Ontology analysis for&nbsp; microarray data</h1>
        </td>
      </tr>

      <tr>
        <td width="72"><!-- Left gutter space -->
        <img height="1" alt="" src="/images/shim.gif" width="70"
        border="0"></td>

        <td valign="top" width="687">
          <h2>Overview</h2>

          <p>ErmineJ performs analyses of <b>gene sets</b> in
          <b>expression microarray data</b>. A typical goal is to
          determine whether particular biological pathways are "doing
          something interesting" in the data. The software is designed
          to be used by biologists with little or no informatics
          background. A command-line interface is available for users
          who wish to script the use of ermineJ.</p>

          <p>Key features include:</p>

          <ul>
            <li>
              Implementation of several methods for gene set analysis: 

              <ul>
                <li><a href=
                "ermineJ-help/html/tutorial/ora.html">Over-representation
                analysis</a></li>

                <li><a href=
                "ermineJ-help/html/tutorial/resampling.html">A
                resampling-based method that uses gene scores</a></li>

                <li>A resampling-based method that uses <a href=
                "ermineJ-help/html/tutorial/correlation.html">correlation
                between gene expression profiles</a> .</li>
              </ul>
            </li>

            <li>Gene sets receive statistical scores (p-values), and
            multiple test correction is supported.</li>

            <li>Support of the Gene Ontology terminology</li>

            <li>User files use <a href="formats">simple text
            formats</a>.</li>

            <li>Users can <a href=
            "ermineJ-help/html/tutorial/classes.html">modify gene sets
            or create new ones</a>.</li>

            <li>The results can be <a href=
            "ermineJ-help/html/tutorial/visualize.html">visualized
            within the software</a>.</li>

            <li>It is simple to compare multiple analyses of the same
            data set with different settings.</li>

            <li>Hyperlinks are provided to external sites to allow more
            efficient browsing of the results.</li>
          </ul>

          <p><a href="ermineJ-help/html/license.html">License
          information</a> - ermineJ is free software.</p>

          <h3>News</h3>

 	  
    <p>October 13 2004 Version 2.0 released. This fixes several GUI bugs that affected use under Linux and Mac, as well
    as some other minor bugs and documentation improvements.</p>

	  
<p>September 28 2004 Version 2.0b7. This further improves
documentation and fixes bugs, including one that caused gene set probe
numbers to be displayed incorrectly when data had been filtered.</p>
    
   <p>September 22 2004  Version 2.0b6 released. This fixes some major bugs in the visualization component of the software, and
   includes some improvements to the documentation and error messages. New feature: some methods for speeding up the
   resampling-based analysis have been implemented. These are particularly helpful for the correlation-analysis. The new methods
   use some approximations that are documented <a href="ermineJ-help/html/tutorial/approximations.html">here</a>.</p>
 
          <p>August 18 2004 - Version 2.0b5 released. This introduces
          some minor new features and fixes several bugs, one of which
          are serious affecting the correlation analysis. In addition,
          the advertised command line tool is now operational.</p>

          <p>August 12 2004 - Version 2.0b4 released. This is a
          maintenance release that fixes a number of bugs and issues,
          some of which are serious. All users should update to this
          version.</p>

 
          <p>July 20 2004 - Version 2.0b3 released. This is an
          extensive rewrite of the user interface, with numerous new
          features, including the ability to define your own gene sets
          and visualize the analysis results. This is a late-beta
          release as we iron out bugs and performance issues.</p>

          <h3>Download and installation</h3>

          <h4>Requirements</h4>

          <p>ermineJ requires a Java 2 Runtime Environment (JRE) 1.4 or
          newer to run. You can download the platform-specific packages
          from SUN's website (<a href=
          "http://java.sun.com/j2se/">http://java.sun.com/j2se/</a> ).
          Users of MS Windows NT/2000/XP can alternatively download the
          Windows installer with bundled JRE package below.</p>

          <h4>Software Files</h4>

          <table width="100%" border="0" cellspacing="1" cellpadding=
          "3" bgcolor="#aaaaaa">
            <tr bgcolor="#eeeeee">
              <td width="37%"><strong>Description</strong></td>

              <td width="30%"><strong>File</strong></td>

              <td width="33%"><strong>Size</strong></td>
            </tr>

            <tr>
              <td bgcolor="#ffffff">Windows installer with bundled
              JRE</td>

              <td bgcolor="#ffffff"><a href=
              "distributions/ermineJ-setup-jre.exe">ermineJ-setup-jre.exe</a></td>

              <td bgcolor="#ffffff">~30 MB</td>
            </tr>

            <tr>
              <td bgcolor="#ffffff">Windows installer without JRE</td>

              <td bgcolor="#ffffff"><a href=
              "distributions/ermineJ-setup.exe">ermineJ-setup.exe</a></td>

              <td bgcolor="#ffffff">~7 MB</td>
            </tr>

            <tr>
              <td bgcolor="#ffffff">Generic bundle</td>
 <td bgcolor="#ffffff"><a href="distributions/ermineJ-2.0-bindist.tar.gz">ermineJ-2.0-bindist.tar.gz</a></td>

              <td bgcolor="#ffffff">~7 MB</td>
            </tr>

            <tr>
              <td bgcolor="#ffffff">Source code</td>

              <td bgcolor="#ffffff"><u><a href=
 
              "distributions/ermineJ-2.0-src.tar.gz"><font color=
              "#800000">ermineJ-2.0-src.tar.gz</font></a></u></td>
 

              <td bgcolor="#ffffff">&lt; 1 MB</td>
            </tr>
          </table>

          <h3>Annotation files</h3>

          <p>To use the software, you have to have microarray
          annotation files and a recent Gene Ontology XML file.</p>

          <p>Go <a href="/annots/">here</a> to get annotation
          files.&nbsp; Unzip the one(s) you need and put them somewhere
          you can find them.</p>

          <p>Go <a href=
          "http://www.godatabase.org/dev/database/archive/latest/">here</a>
          to get the XML file (something like <a href=
          "http://www.godatabase.org/devdatabase/archive/latest/go_200409-termdb.xml.gz">http://www.godatabase.org/devdatabase/archive/latest/go_200409-termdb.xml.gz</a>; the 
          date portion of the name may have changed).
          Unzip it and put it somewhere you can locate it later.</p>

          <h3>Installation</h3>

          <p>On MS Windows NT/2000/XP, simply run the installer.</p>

	  <p>For other platforms, you should be able to use the
"Generic bundle". There is a README file included that gives some
instructions.</p>

          <h3>WebStart</h3>

          <p>You can <a href="webstart">run ermineJ</a> from any
          operating system that supports Java (in theory) without
          running the installer by using java webstart.</p>

          <p>This a good way to get the latest version, as we will keep
          the link updated with the most recent development build. On
          the other hand, this version should be considered less stable
          than the other releases. This is here partly for our own
          testing purposes.</p>

          <h3>Documentation</h3>

          <p>The software comes with on-line help. The manual is also
          available <a href="ermineJ-help/html">here</a> .</p>
          <hr>

          <h3>Contact</h3>

          <p>Contact <a href="mailto:pavlidis@dbmi.columbia.edu">Paul
          Pavlidis</a> with bug reports, feature requests, and
          questions.</p>

          <h3>Publications using the software or methods</h3>

          <ul>
            <li><!--StartFragment -->Sibille, E., Arango, V., Galfalvy,
            H.C., Pavlidis, P., Ellis, S.P. and Mann, J.J. (2004)
            Genomic analysis of depression and suicide in human
            prefrontal cortex. Neuropsychopharmacology 29:351-361
            .</li>

            <li><!--StartFragment -->&nbsp;Pavlidis P., Qin J, Arango
            V, Mann J, Sibille E. (2004) Using the Gene Ontology for
            microarray data mining: A comparison of methods and
            application to age effects in human prefrontal cortex.
            Neurochemical Research, 29(6) 1213-1222&nbsp;.</li>

            <li>Pavlidis, P., Qin, J and Sibille E (2003) Statistical
            analysis of Gene Ontology classes as tools for
            understanding gene expression changes in the brain. Society
            for neuroscience Annual Meeting Abstract 758.12. (<a href=
            "neurosci2003.ppt"><font color="#0000ff">download the
            powerpoint version of the poster</font></a> )</li>

            <li>Pavlidis, Lewis and Noble, Proceedings of the Pacific
            Symposium on Biocomputing 7. pp 474-485, 2002. (<a href=
            "/pavlidis/doc/reprints/pavlidis-psb2002.pdf"><font color=
            "#0000ff">Download a reprint</font></a> .</li>
          </ul>

          <h3>Credits</h3>

          <p>The software was written by Paul Pavlidis, Homin Lee and
          William Braynen. Some code is based on an earlier version of
          the software written by Shahmil Merchant and Edward Chen,
          based in turn on a Perl implementation by Paul Pavlidis. Our
          Gene Ontology annotations were obtained through the <a href=
          "http://www.geneontology.org/">Gene Ontology Consortium web
          site</a>.</p>
        </td>
      </tr>
    </table>
  </body>
</html>
