<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>
<head>
	<title>ErmineJ</title>
	<link rel="stylesheet" href="css/default.css">
</head>

<body bgcolor="#FFFFFF" link="#800000" text="#000000">
<table border="0" cellpadding="2" cellspacing="3" width="600">
<tr>
	<td>
	<!-- Logo space -->
		<a href="index.html">Home</a>
		
	</td>
	<td>
	<!-- Header space -->
	<h1>ErmineJ: Requirements</h1>
	</td>
</tr>
<tr>
	<td>
	<!-- Left gutter space -->
	<img src="/images/shim.gif" width="70" height="1" border="0" alt="">

	</td>
	<td>
	<!-- main text space -->
	<p><strong>This is out of date!</strong></p>
	<p>This document provides only a user-level requirement outline and tries to stay away from any issues of implementation or system
	architecture, which will be described in other documents.</p>
	<h2>Project Overview</h2>
	<p>ErmineJ (the name is provisional) will be a software tool for gene expression data analysis. It will be implemented in Java and include
	a graphical user interface.</p>
	<p>The target platform is Windows, with unix (solaris/linux) and Macintosh compatibility being secondary in consideration.</p>
	
	<p>ErmineJ will encompass two main functionalities: <b>Analysis</b> and <b>Visualization/Browsing</b>. The overall goal, however
	is to provide a software framework for expansion of the functionality. The initial goals are aimed at implementing methods
	we have currently been running in Unix from the command line or via web browser interfaces, implmeneted in C and Perl.</p>
	
	<div class="indent">
	<h3>Analysis section </h3>
	
	<p>The analysis aspect of ErmineJ is aimed at the tools we have developed which are not available elsewhere, rather than
	reimplementing the wheel. However, some simple "redundant" tools are going to be necessary to make this really a useful tool. These include:</p>
	<ul>
	<li><a href="http://microarray.cpmc.columbia.edu/pavlidis/protected/ermgene/web/">Class scoring (ErmGene)</a>. This will be the initial priority.</li>
	<li>Template matching (this is available in other tools, such as Genespring, but we can provide many enhancements)</li>
	<li>T-tests and one-way ANOVA (or at least initially, import of the pvalues).</li>
	</ul>
	
	<h3>Visualization/Browsing</h3>
	<p>This will provide a point-and-click interface to simple &quot;block&quot; visualizations such as those provided by 
	<a href="http://microarray.cpmc.columbia.edu/pavlidis/pub/matrix2png/web/">matrix2png</a>. This will include a GO browser as well.</p>
	</div>
	
	<p>Further details (in overview) of requirements for these two aspects follow. To be determined is the overall design of the system; will the visualization
	live in a separate window frame from the analysis? Or will they coexist with a common set of menus etc.? Prioritization of
	the requirements also is to be determined.</p>
	
	<h3>Analysis</h3>
	<h4>Reading data files</h4>
	
	<p>The analysis software must be able to 
	read data files in the simple text format specified <a href="http://microarray.cpmc.columbia.edu/pavlidis/pub/stats/web/dataformat.html">here</a>.
	Parsing of <a href="http://genome-www.stanford.edu/~sherlock/cluster.html#formats">cluster</a> format files must be available as well.</p>
	
	<p>The software must load the annotations files provided <a href="http://microarray.cpmc.columbia.edu/pavlidis/protected/arraytools/web/classes.html">here</a> (gene
	ontology classes)
	and <a href="http://microarray.cpmc.columbia.edu/fs/databases/affymetrix/annotsupdate.html">here</a> (gene annotations for genebank etc.). The files are
	found in either a user-configued location or in a default location. Loading of these files should happen 'behind the scenes' but the information
	in them is available to the user (via the interface) and analysis methods.</p>
	
	<p>Template matching: Reading Template files: the format for these is specified <a href="http://microarray.cpmc.columbia.edu/pavlidis/pub/stats/web/define-templates.html">here</a>.
	Not described there is the 'multi-template file', which is similar except that there is one template per line and there is an additional column
	of template names (the first column).</p>
	
	<p>Reading layout and classification files: These files are described <a href="http://microarray.cpmc.columbia.edu/pavlidis/pub/stats/web/layout.html">here</a> (layouts)
	and here (classification files) and describe the various categories of samples present.</p>
	
	<p>Reading p-value files: The software must open and read score files provided by the user. These will consist of one column of gene names (probe set ids
	typically) and one column of scores. A header line will be present optionally</p>
	
	<h4>Output</h4>
	<p>The analysis software must provide text output in the same 
	data input <a href="http://microarray.cpmc.columbia.edu/pavlidis/pub/stats/web/dataformat.html">format</a> mentioned above
	(i.e., for data from specific gene classes etc.). The option of 
	exporting <a href="http://genome-www.stanford.edu/~sherlock/cluster.html#formats">cluster</a> format files must be available. This
	way the output can be used in the Stanford cluster software. (visualization output issues are described elsewhere)</p>
	
	<p>Pvalues output: The pvalues or other analysis results are exported in a text file.</p>
	
	<p>Score distribution output: Background score histograms are exported in a text file.</p>
	
	<p>Preferences: the settings used in the analysis are saved in a file and read in at the beginning of a session.</p>
	
	<h4>Analysis methods</h4>
	<p>The primary goal will be to implement the "correlation" 
	and "experiment" gene class score methods described in the paper 
	found <a href="http://microarray.cpmc.columbia.edu/pavlidis/doc/reprints/pavlidis-psb2002.pdf">here</a>. The latter requires the use or generation
	of gene-wise pvalues. Probably initially we will import pvalues that the user provides. Later we would like some simple methods of generating pvalues
	or using pvalues generated by other tools in the system such as template matching.</p>
	
	<p>For the experiment score, the user will have the option of using "median" or other quantile instead of "mean" for calculating raw class scores.</p>
	
	<p>Both methods are computationally intensive, as they have to calculate empirical background distributions; the software should provide a quick
	heuristic method (or even raw scores) so that a quick result can be viewed before deciding to calculate the full distributions.</p>
	
	<p>Display of the results in the visualization, organized by GO classification. Ideally we will show a classification tree. The
	mechanism for the display must be sufficiently flexible to show any classification scheme (including those which are not really trees) and
	not be restricted to GO.</p>
	
	<h4>Database connectivity</h4>
	<p>In the long term, the software must be capable of getting data from a database, and storing the results in a database, and providing output
	in XML or other database-readable formats (MAGE-ML etc.) as appropriate.</p>
	
	<h4>User interface</h4>
	<p>status bar: A status line is provided to show the state of the program.</p>
	<p>progress screens: during lengthy tasks, a progress indicator is shown (perhaps on the status bar)</p>
	<p>menu driven: All functions are driven by menus or toolbar buttons.</p>
	<p>web links: HTML links are provided to all gene annotations so that a web browser can be invoked automatically
	to view publically-available details about genes.</p>
	<p>tables: in some cases, the most efficient way to display numerical data is in a table. This will be used to show ranked lists of classes.
	by pvalue.</p>
	
	<h3>Example of a how a user session might go</h3>
	<ol>
	<li>User starts the software, sees a splash screen announcing the initiation of the program</li>
	<li>The software loads any initiation files; a progress screen is shown if this takes longer than a couple of seconds</li>
	<li>The user selects "file...open data" to select a data file to read in. A list of recently used files is presented to make this easier.</li>
	<li>The user selects "analysis...class scoring" and is presented with a dialog box with the settings for the class scoring analysis.</li>
	<li>The user selects the appropriate options and clicks "run analysis". A progress bar is shown.</li>
	<li>When the analysis is shown, a visualization of the results is automatically shown.</li>
	<li>The user selects a class of interest in the visualization with a context-specific-menu selects "save data".</li>
	</ol>
	
	<h3>Jotted notes</h3>
	<pre>
	web links for everything
	stubs for database connectivity
	display of go tree as a 'file tree'
	plotting gene graphs
	output: postscript, png for everything graphical. (very important!)
	templates: choose a gene; compare to other datasets.
	mechanism for adding data sets for template matching.
	mechanism for designing templates.
	</pre>
	
	<h3>References</h3>
	
<div class="reference">	<dl>
	<dt>--
	</dl></div>
	</td>
</tr>
<tr name="footer">
<td colspan="2"><img src="/images/shim.gif" width="1" height="10" border="0" alt=""></td>
</tr>
<tr>
<td>
<!-- left gutter psace -->
</td>
<td class="footer"><a href="mailto:pp175@columbia.edu">Paul Pavlidis</a>.&nbsp; 
<script language="JavaScript" type="text/javascript">
<!--
	document.write("Last Modified " + document.lastModified);
// -->
</script>
</td>
</tr>
</table>

</body>
</html>
